src/data/database.py

```
@@ -9,35 +9,74 @@
-engine = create_engine(
-    DATABASE_URL,
-    pool_pre_ping=True,  # Check connection before using
-    pool_size=5,  # Default pool size
-    max_overflow=10,  # Allow up to 10 additional connections
-    pool_recycle=3600,  # Recycle connections after 1 hour
-)
+if _env_database_url:
+    DATABASE_URL = _env_database_url
+else:
+    # Safely build Postgres URL if all parts are provided
+    if DB_USER and DB_PASSWORD and DB_NAME:
```

Ruff (0.12.2)

Use `elif` instead of `else` then `if`, to reduce indentation

Convert to `elif`

(PLR5501)